addons/binding/org.openhab.binding.homie/src/main/java/org/openhab/binding/homie/internal/conventionv200/TopicParser.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homie.internal.conventionv200;
+
+import static org.openhab.binding.homie.internal.conventionv200.HomieConventions.ID_PATTERN;
+
+import java.text.ParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openhab.binding.homie.HomieBindingConstants;
+
+/**
+ * Parser for Homie MQTT Topics
+ *
+ * @author Michael Kolb - Initial Contribution
+ *
+ */
+public class TopicParser {
+
+    protected final static String MATCHGROUP_INTERNAL_PROPERTY_NAME = "intproperty";
+    protected final static String MATCHGROUP_INTERNAL_SUBPROPERTY_NAME = "intsubproperty";
+    protected final static String MATCHGROUP_PROPERTY_NAME = "property";
+    protected final static String MATCHGROUP_NODEID_NAME = "nodeid";
+    protected final static String MATCHGROUP_DEVICEID_NAME = "deviceid";
```

Comment:
I am aware that this violates the common java constant naming pattern, but for a good reason:
Currently it is clear that the ID is related to the device. If we change it to MATCHGROUP_DEVICE_ID_NAME: Is it now a Device-Id or a Device with an ID-Name?